### PR TITLE
Postgres migration

### PR DESCRIPTION
**Pull request by friendkapuzio**
add OAuthRegistrationRestriction and repo, update OAauthReg and scopes

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/entity/oauth/OAuthRegistrationRestriction.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import java.util.Objects;
+
+@Entity
+@EntityListeners(AuditingEntityListener.class)
+@Table(name = "oauth_registration_restriction", schema = "public")
+public class OAuthRegistrationRestriction implements Serializable {
+	@Id
+	@GeneratedValue(strategy = GenerationType.IDENTITY)
+	@Column(name = "id", unique = true, nullable = false, precision = 64)
+	private Integer id;
```
Comment (Yumfriez) [inline src/main/java/com/epam/ta/reportportal/entity/oauth/OAuthRegistrationRestriction.java]: Replace Integer id type with Long

Comment (Yumfriez) [reply src/main/java/com/epam/ta/reportportal/entity/oauth/OAuthRegistrationRestriction.java]: Remove precision

Comment (pbortnik) [reply src/main/java/com/epam/ta/reportportal/entity/oauth/OAuthRegistrationRestriction.java]: you could remove all these restrictions as unique and nullable cause we decided leave all that on db side

File: src/main/java/com/epam/ta/reportportal/dao/OAuthRegistrationRestrictionRepositoryCustom.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.epam.ta.reportportal.entity.oauth.OAuthRegistrationRestriction;
+
+import java.util.List;
+
+/**
+ * @author Anton Machulski
+ */
+public interface OAuthRegistrationRestrictionRepositoryCustom {
+	List<OAuthRegistrationRestriction> findByRegistrationId(String registrationId);
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/dao/OAuthRegistrationRestrictionRepositoryCustom.java]: i suppose such queries could be done with existed jpa query builder
https://docs.spring.io/spring-data/jpa/docs/current/reference/html/#repositories.query-methods.query-lookup-strategies

File: src/main/java/com/epam/ta/reportportal/dao/OAuthRegistrationRestrictionRepositoryCustomImpl.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+public class OAuthRegistrationRestrictionRepositoryCustomImpl implements OAuthRegistrationRestrictionRepositoryCustom {
+	private DSLContext dsl;
+
+	@Autowired
+	public void setDsl(DSLContext dsl) {
+		this.dsl = dsl;
+	}
+
+	@Override
+	public List<OAuthRegistrationRestriction> findByRegistrationId(String registrationId) {
```
Comment (pbortnik) [inline src/main/java/com/epam/ta/reportportal/dao/OAuthRegistrationRestrictionRepositoryCustomImpl.java]: move to jpa

**Pull request merged**